OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs

```diff
@@ -127,6 +127,21 @@ public EditorActorPreview Add(string id, ActorReference reference, bool initialS
 				list.Add(preview);
 			}
 
+			// Fallback to the actor's CenterPosition for the ActorMap if it has no Footprint
+			if (!preview.Footprint.Any())
+			{
+				var location = worldRenderer.World.Map.CellContaining(preview.CenterPosition);
+
+				List<EditorActorPreview> list;
```

Comment:
Could consider to extract the following section to remove the duplication.

Reply:
I don't think it is really necessary, but done.